io-engine/src/lvm/lv_replica.rs

```diff
@@ -0,0 +1,739 @@
+        Self {
+            lv,
+            ..self
+        }
+    }
+    /// Get a comma-separated list of query selection args.
+    /// todo: should be Display trait?
+    pub(super) fn query(&self) -> String {
+        let mut select = self.vg.query();
+        let args = &self.lv;
+
+        if self.regular_lv {
+            if let Some(lv_name) = &args.name {
+                select.push_str(&format!("lv_name={lv_name},"));
+            }
+            if let Some(lv_uuid) = &args.uuid {
+                select.push_str(&format!("lv_uuid={lv_uuid},"));
+            }
+        } else {
+            if let Some(name) = &args.name {
+                select.push_str(&format!(
+                    "lv_tags={},",
+                    Property::LvName(name.to_string()).tag()
+                ));
+            }
+            if let Some(lv_name) = &args.uuid {
+                select.push_str(&format!("lv_name={lv_name},"));
+            }
+        }
+
+        if let Some(lv_tag) = &args.tag {
+            select.push_str(&format!("lv_tags={lv_tag},"));
+        }
+        select
+    }
+}
+
+/// A Logical Volume (LV) is a virtual block device that can be used by the
+/// system or applications.
+/// Each block of data in an LV is stored on one or more PV in the VG, according
+/// to algorithms implemented by Device Mapper (DM) in the kernel.
+#[derive(Debug, Clone, Deserialize)]
+pub struct LogicalVolume {
+    /// The LV uuid.
+    /// This is generated by LVM and we cannot specify a particular value.
+    #[allow(dead_code)]
+    lv_uuid: String,
+    /// The LV name.
+    /// We set this to the Replica UUID.
+    lv_name: String,
+    /// The name of the parent VG.
+    vg_name: String,
+    /// The uuid of the parent VG.
+    /// This is generated by LVM and we cannot specify a particular value.
+    vg_uuid: String,
+    #[serde(deserialize_with = "de::number_from_string")]
+    vg_extent_size: u64,
+    /// The LV path.
+    /// The logical volume full path name is /dev/<vg_name>/<lv_name>.
+    #[serde(rename = "lv_path")]
+    path: String,
+    /// Size of the LV.
+    #[serde(rename = "lv_size")]
+    #[serde(deserialize_with = "de::number_from_string")]
+    size: u64,
+    /// Tags of the LV converted to our properties.
+    #[serde(rename = "lv_tags")]
+    #[serde(deserialize_with = "de::comma_separated")]
+    tags: Vec<Property>,
+
+    #[serde(skip)]
+    runtime: RunLogicalVolume,
+}
+
+/// Runtime settings for the LogicalVolume.
+#[derive(Debug, Default, Clone)]
+pub struct RunLogicalVolume {
+    /// The protocol of the SPDK Bdev which is created against the lv_path.
+    share: Protocol,
+    /// LV's are created with name as a tag, so we have to probe the tags and
+    /// set name if we find the tag.
+    name: Option<String>,
+
+    /// SPDK Bdev parameters which are needed by LVM.
+    bdev: Option<BdevOpts>,
+}
+
+/// Runtime settings for the LogicalVolume.
+#[derive(Debug, Default, Clone)]
+pub struct BdevOpts {
```

Comment:
Would this be better renamed and moved to common place where we already have `BdevOpts` and `NvmeBdevOpts` ?